include/vcpkg/base/messages.h

```diff
@@ -892,6 +894,15 @@ namespace vcpkg
                     "",
                     "invalid argument: binary config '{binary_source}' requires a SAS token without a "
                     "preceeding '?' as the second argument");
+    DECLARE_MESSAGE(InvalidCommandArgMaxDepth, (), "", "Value of --max-depth must be an integer.");
```

Comment:
Does it make sense to make this one reusable? e.g.

DECLARE_MESSAGE(OptionMustBeInteger, (msg::option), "", "Value of --{option} must be an integer.");

(Maybe consider alongside your vcpkgcmdarguments.cpp PR)